cpp/src/decisiontree/memory.cuh

```diff
@@ -25,35 +25,26 @@ template <class T, class L>
 TemporaryMemory<T, L>::TemporaryMemory(
   const std::shared_ptr<MLCommon::deviceAllocator> device_allocator_in,
   const std::shared_ptr<MLCommon::hostAllocator> host_allocator_in,
-  const cudaStream_t stream_in, int N, int Ncols, float colper, int n_unique,
-  int n_bins, const int split_algo, int depth, bool col_shuffle) {
+  const cudaStream_t stream_in, int N, int Ncols, int n_unique,
+  const ML::DecisionTree::DecisionTreeParams& tree_params) {
   stream = stream_in;
-  splitalgo = split_algo;
```

Comment:
It does not seems to me that member variable `splitalgo` is used anywhere so removing it.